ws-android/Dockerfile

```diff
@@ -27,43 +38,50 @@ RUN SDK_VERSION=r24.3.3 && \
+ENV ANDROID_HOME /usr/local/android-sdk-linux
+ENV PATH $PATH:$ANDROID_SDK_HOME
+ENV PATH $PATH:$ANDROID_SDK_HOME/tools
+ENV PATH $PATH:$ANDROID_SDK_HOME/platform-tools
+
+# -------------------------- Install the Android Platforms -----------------------------------------
+
+# This fancy "for" loop solves an install issue
+RUN for filter in tools platform-tools android-20 android-22 android-23 extra build-tools-23.0.2; do \
+        echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter $filter --no-ui --force --all; \
+    done
+
+
+
+# -------------------------- Install Gradle -----------------------------------------
+
+ENV GRADLE_VERSION 2.9
+
+RUN cd /usr/local && \
     curl -L https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip -o gradle-bin.zip && \
     unzip gradle-bin.zip && \
-    rm gradle-bin.zip
-
-# Update & Install Android Tools
-# Cloud message, billing, licensing, play services, admob, analytics
-RUN yes | (\
-    /usr/local/android-sdk-linux/tools/android update sdk --filter tools --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter platform-tools --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter android-19 --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter android-20 --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter android-21 --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter android-22 --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter android-23 --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --filter extra --no-ui --force -a && \
-    /usr/local/android-sdk-linux/tools/android update sdk --all --filter build-tools-23.0.2 --no-ui --force) 
-
-# Set PATH
-RUN \
-    echo 'export ANDROID_HOME=/usr/local/android-sdk-linux' > /etc/profile.d/android.sh && \
-    echo 'export ANDROID_SDK_HOME=/usr/local/android-sdk-linux' >> /etc/profile.d/android.sh && \
-    echo 'export ANDROID_NDK_HOME=/usr/local/android-ndk-r10e' >> /etc/profile.d/android.sh && \
-    echo 'export GRADLE_HOME=/usr/local/gradle-2.5' >> /etc/profile.d/android.sh && \
-    echo 'export NDK_PROJECT_PATH=/home/ubuntu/workspace' >> /etc/profile.d/android.sh && \
-    echo 'export PATH=$PATH:$ANDROID_HOME/tools:$ANDROID_NDK_HOME/platform-tools:$ANDROID_NDK_HOME:$GRADLE_HOME/bin' >> /etc/profile.d/android.sh
+    rm gradle-bin.zip  && \
+    chmod -R +r /usr/local/gradle-${GRADLE_VERSION} && \
+    bash -c 'for file in $(find /usr/local/gradle-${GRADLE_VERSION}); do [[ -x $file ]] && chmod +x "$file"; done'; \
+    exit 0
+
+#Gradle Home and Path for pre-cloud9 testing
+ENV GRADLE_HOME /usr/local/gradle-${GRADLE_VERSION}
+ENV PATH $PATH:$GRADLE_HOME/bin
+
+
+# activate gradle by running it once    
+RUN gradle
```

Comment:
if you call it from a bash login with `RUN bash -l -c "gradle"` then it will use the values we already put in profile.d, and we don't have to maintain them in two places